### PR TITLE
fs/vfs: Implement statvfs and fstatvfs

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
specified here:
https://pubs.opengroup.org/onlinepubs/009695399/basedefs/sys/statvfs.h.html

## Impact
Add statvfs/fstatvfs, but no other code change.

### CODE REVIEW

File: arch/sim/src/sim/up_hostfs.c [hunk 1]
```diff
@@ -432,32 +432,35 @@ int host_closedir(void *dirp)
...
       ret = -errno;
     }
 
-  /* Map the struct statfs value */
+  /* Map the struct statvfs value */
 
-  buf->f_type    = 0; /* hostfs overwrites f_type anyway */
-  buf->f_namelen = hostbuf.f_namemax;
+  buf->f_fsid    = 0; /* hostfs overwrites f_fsid anyway */
+  buf->f_flag    = hostbuf.f_flag;
```
Comment (yamt) [inline arch/sim/src/sim/up_hostfs.c:454]: f_flags needs host->nuttx conversion. maybe it's safer to use 0 for now.

File: include/sys/statvfs.h [hunk 2]
```diff
@@ -49,7 +49,14 @@
  * Pre-processor Definitions
  ****************************************************************************/
 
-/* struct statfs file system types. */
+/* Definitions for the flag in `f_flag'.  These definitions should be
+ * kept in sync with the definitions in <sys/mount.h>.
+ */
+
+#define ST_RDONLY             0x0001 /* Mount read-only.  */
+#define ST_NOSUID             0x0002 /* Ignore suid and sgid bits.  */
```
Comment (yamt) [inline include/sys/statvfs.h:57]: we don't have NOSUID things.

File: arch/sim/src/sim/up_hostfs.c (hunk 1 above)
Comment (xiaoxiang781216) [reply arch/sim/src/sim/up_hostfs.c:454]: Done.

File: include/sys/statvfs.h (hunk 2 above)
Comment (xiaoxiang781216) [reply include/sys/statvfs.h:57]: Yes, but it isn't harmful to keep it here since we may imlement this feature someday.

Comment (yamt) [reply include/sys/statvfs.h:57]: ok

**Comment by xiaoxiang781216** [comment]
@yamt do you have more concern about the change? If no, please merge this patch and apache/incubator-nuttx-apps#325. The check failure has to be ignored since the change in nuttx and apps need sync to pass the compile.

**Comment by yamt** [comment]
@xiaoxiang781216 i think f_fsid is meant to be an ID for each individual file system, not file system type. how do you think?

**Comment by xiaoxiang781216** [comment]
> @xiaoxiang781216 i think f_fsid is meant to be an ID for each individual file system, not file system type. how do you think?

Yes, you are right. After more thinking, I will keep both interfaces to improve the compatibility.

**Comment by xiaoxiang781216** [comment]
@ymat, the new patch implement statvfs/fstatvfs on top of statfs/fstatfs. Please review again.

File: include/sys/statvfs.h [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+ ****************************************************************************/
+
+#ifndef __INCLUDE_SYS_STATVFS_H
+#define __INCLUDE_SYS_STATVFS_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <sys/statfs.h>
```
Comment (yamt) [inline include/sys/statvfs.h:28]: is this include necessary?

Comment (xiaoxiang781216) [reply include/sys/statvfs.h]: Not necessary, I changed to sys/types.h

Comment (xiaoxiang781216) [reply include/sys/statvfs.h]: Done, @yamt.

**Pull request merged**